app/controllers/spree/admin/payments_controller.rb

```diff
@@ -142,6 +143,17 @@ def can_transition_to_payment
         redirect_to spree.edit_admin_order_customer_url(@order)
       end
 
+      def ensure_sufficient_stock_lines
+        return if @order.insufficient_stock_lines.blank? || !@order.payment?
```

Comment:
Great.

I'm wondering if `!@order.payment?` here should be `@order.competed?` instead?

Also; with these two parts of the conditional; the first part (stock check with variant override scoping) is expensive and the second part is really cheap. If we switch the order, we can fail early in lots of cases and avoid running the stock check entirely, eg:
```ruby
return if @order.competed? || @order.insufficient_stock_lines.blank? 
```

Reply:
```suggestion
        return if !@order.payment? || @order.insufficient_stock_lines.blank?
```